ksqldb-engine/src/test/java/io/confluent/ksql/function/udaf/offset/EarliestByOffsetUdafTest.java

```diff
@@ -23,39 +23,110 @@
 import static io.confluent.ksql.function.udaf.KudafByOffsetUtils.STRUCT_STRING;
 import static io.confluent.ksql.function.udaf.KudafByOffsetUtils.VAL_FIELD;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 
+import com.google.common.collect.Lists;
 import io.confluent.ksql.function.udaf.Udaf;
+import io.confluent.ksql.util.KsqlException;
+import java.util.ArrayList;
+import java.util.List;
 import org.apache.kafka.connect.data.Struct;
 import org.junit.Test;
 
+
 public class EarliestByOffsetUdafTest {
+  
   @Test
   public void shouldInitialize() {
     // Given:
-    final Udaf<Integer, Struct, Integer> udaf = EarliestByOffset
-        .earliest(STRUCT_LONG);
+    final Udaf<Integer, Struct, Integer> udaf = EarliestByOffset.earliest(STRUCT_LONG);
 
     // When:
     Struct init = udaf.initialize();
 
     // Then:
     assertThat(init, is(notNullValue()));
   }
+  
+  @Test
+  public void shouldInitializeN() {
+    // Given:
+    final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset.earliestN(STRUCT_LONG, 2);
+
+    // When:
+    List<Struct> init = udaf.initialize();
+
+    // Then:
+    assertThat(init, is(notNullValue()));
+  }
+  
+  @Test
+  public void shouldThrowExceptionForInvalidN() {
+    try {
+      final Udaf<Integer, List<Struct>, List<Integer>> udaf = EarliestByOffset
```

Comment:
Fix spot bugs issue:

```suggestion
      EarliestByOffset
```